transformers4rec/torch/utils/data_utils.py

```diff
@@ -170,187 +182,180 @@ def from_schema(
-
-            self.paths_or_dataset = paths_or_dataset
-            self.batch_size = batch_size
-            self.shuffle = shuffle
-            self.max_sequence_length = max_sequence_length
-            self.drop_last = drop_last
+class DLDataLoader(PyTorchDataLoader):
+    """
+    This class is an extension of the torch dataloader.
+    It is required to support the FastAI framework.
 
-            self.set_dataset(buffer_size, engine, reader_kwargs)
+    Setting the batch size directly to DLDataLoader makes it 3x slower.
+    So we set as an alternative attribute and use it within
+    T4Rec Trainer during evaluation
+    # TODO : run experiments with new merlin-dataloader
+    """
 
-            self.dataset.schema = _augment_schema(
-                self.dataset.schema, cats, conts, labels, sparse_names, sparse_max, sparse_as_dense
-            )
+    def __init__(self, *args, **kwargs) -> None:
+        if "batch_size" in kwargs:
+            self._batch_size = kwargs.pop("batch_size")
+            super().__init__(*args, **kwargs)
 
-            if (global_rank is not None) and (self.dataset.npartitions < global_size):
-                logger.warning(
-                    "UserWarning: User is advised to repartition the parquet file before training "
-                    "so npartitions>=global_size. Cudf or pandas can be used for repartitioning "
-                    "e.g.: df.to_parquet('file.parquet', row_group_size=N_ROWS/NPARTITIONS, engine"
-                    "='pyarrow') as npartitions=nr_rows/row_group_size."
-                )
-                self.dataset = self.dataset.repartition(npartitions=global_size)
+    @property
+    def device(self):
+        return torch.device("cuda" if torch.cuda.is_available() else "cpu")
 
-            loader = Loader(
-                self.dataset,
-                self.batch_size,
-                shuffle,
-                seed_fn=seed_fn,
-                parts_per_chunk=parts_per_chunk,
-                device=device,
-                global_size=global_size,
-                global_rank=global_rank,
-                drop_last=drop_last,
-            )
+    def __len__(self):
+        return len(self.dataset)
 
-            DLDataLoader.__init__(
-                self,
-                loader,
-                collate_fn=collate_fn,
-                batch_size=self.batch_size,
-                drop_last=self.drop_last,
-            )
-            self.schema = schema
-            self.max_sequence_length = max_sequence_length
 
-        def set_dataset(self, buffer_size, engine, reader_kwargs):
-            dataset = validate_dataset(
-                self.paths_or_dataset,
-                self.batch_size,
-                buffer_size,
-                engine,
-                reader_kwargs,
+@dataloader_registry.register_with_multiple_names(
+    "merlin_dataloader", "merlin", "nvtabular_dataloader", "nvtabular"
+)
+class MerlinDataLoader(T4RecDataLoader, DLDataLoader):
```

Comment:
I removed the condition `if dependencies.is_gpu_dataloader_available()` to always define the MerlinDataLoader (for cpu + gpu based environments)